lib/rouge/lexers/csound.rb

```diff
@@ -0,0 +1,503 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    # @abstract
+    class Csound < RegexLexer
+      def self.identifier
+        /[A-Z_a-z]\w*/
+      end
+
+      state :whitespace do
+        rule /[ \t]+/, Text
+        rule %r(/\*.*?\*/)m, Comment::Multiline
+        rule %r((?:;|//).*$), Comment::Single
+        rule(/(\\)(\n)/) { groups Text::Whitespace, Text }
+      end
+
+      state :preprocessor_directives do
+        rule /#(?:e(?:nd(?:if)?|lse)\b|##)|@@?[ \t]*\d+/, Comment::Preproc
+        rule /#include/, Comment::Preproc, :include_directive
+        rule /#[ \t]*define/, Comment::Preproc, :define_directive
+        rule /#(?:ifn?def|undef)\b/, Comment::Preproc, :macro_directive
+      end
+
+      state :include_directive do
+        mixin :whitespace
+        rule /([^ \t]).*?\1/, Str, :pop!
+      end
+
+      state :define_directive do
+        rule /\n+/m, Text
+        mixin :whitespace
+        rule /(#{Csound.identifier})(\()/ do
+          groups Comment::Preproc, Punctuation
+          goto :macro_parameter_name_list
+        end
+        rule Csound.identifier do
+          token Comment::Preproc
+          goto :before_macro_body
+        end
+      end
```

Comment:
Do you mean the states? If so, I can separate these, but the states are grouped intentionally. `:define_directive`, `:macro_parameter_name_list`, `:before_macro_body`, and `:macro_body` all deal with defining macros. `:macro_uses`, `:macro_parameter_value_list`, `:macro_parameter_value_quoted_string`, _etc._ all deal with using macros.